### DISCUSSION TITLE
[IntelliSense] reusing styles by declaring as string and using clsx (intellisense support needed)

### DISCUSSION THREAD

**Discussion by GorvGoyl** [tailwindlabs/tailwindcss#9952] [Ideas]
I went through [Reusing Styles](https://tailwindcss.com/docs/reusing-styles) docs, but the mentioned solutions expect extracting the whole component with html or using a CSS file and using `@apply`. I've got one more suggestion :

We can reuse tailwind classes by declaring those as a string and then use that in place of `className="{string}"`. There's a sweet library [clsx](https://www.npmjs.com/package/clsx) that does that and also supports [constructing className strings conditionally](https://blog.makerx.com.au/extracting-reusable-tailwind-react-components/).

rough example: 

```ts
  const primaryButtonStyles = clsx(
    'text-xs text-slate-600 uppercase font-bold tracking-wider',
    props.variantColor === 'red' && 'bg-red-600 text-white'
  )

  const primaryButton = <button className={primaryButtonStyles}>Submit</button>
```

But, the issue is we don't get autocomplete/intellisense when writing tw classes in `clsx()` function in VSCode. Could we please get the intellisense support for it in the VSCode extension?

**Comment by wongjn** [comment]
There's an experimental setting you can use at the moment for your `clsx` usage: https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659

**Reply by GorvGoyl** [reply]
Thanks.. looks like below works:

```
"tailwindCSS.experimental.classRegex": [
    "clsx([^]*)"
  ],
```
